src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -110,6 +207,46 @@ class Distributor
   {
   }
 
+  template <typename View>
+  size_t createFromSends(View const &batched_destination_ranks,
+                         View const &batch_offsets)
```

Comment:
Is it not possible at all to have a single `createFromSends` function where the second argument can have a default value, so we can differentiate inside?

Reply:
In this case, we are assuming that the ranks are already sorted, while in the other function this is not necessarily the case. Of course, we can just check if `batch_offsets` is empty and call `sortAndDetermineBufferLayout` in that case and `DetermineBufferLayout` otherwise. For me, having separate interfaces appears to be cleaner, though.

I am happy to put everything after `determineBufferLayout` in a separate function that can be used for both overloads, though, if you prefer that.

Reply:
I'll leave this up to @dalg24. Personally, I would probably merge them together in a manner you describe as it leads to less code. But I see your point, too.